tests/e2e/p/staking_rewards.go

```diff
@@ -269,9 +269,24 @@ var _ = ginkgo.Describe("[Staking Rewards]", func() {
 		ginkgo.By("determining expected validation and delegation rewards")
 		currentSupply, _, err := pvmClient.GetCurrentSupply(e2e.DefaultContext(), constants.PrimaryNetworkID)
 		require.NoError(err)
-		calculator := reward.NewCalculator(rewardConfig)
-		expectedValidationReward := calculator.Calculate(validationPeriod, weight, currentSupply)
-		potentialDelegationReward := calculator.Calculate(delegationPeriod, weight, currentSupply)
+		expectedValidationReward := reward.Calculate(
```

Comment:
> Why is it desirable to move to a static function?

My thought was that this is entirely stateless which is why I thought it should be static + the interface is not an abstraction that we need. 

> why does that suggest providing these values individually to Calculate instead of via a struct (especially since they are almost always consumed via RewardConfig)?

My thoughts here are that passing these values explicitly makes it harder for subtle bugs to pop up. One example is [here](https://github.com/ava-labs/avalanchego/pull/2062/files#r1334103464) where a test passes with a non-initialized reward config because the reward calculator's reward config is initialized.